cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApi.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2017-2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.assembly;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public interface KafkaConnectApi {
+    Future<JsonObject> createOrUpdatePutRequest(String connectorName, JsonObject configJson);
+    Future<Void> delete(String connectorName);
+    Future<List<String>> list();
+}
+
+class ConnectRestException extends RuntimeException {
+    public ConnectRestException(String message) {
+        super(message);
+    }
+}
+
+@SuppressWarnings({"deprecation"})
+class KafkaConnectApiImpl implements KafkaConnectApi {
+    private static final Logger log = LogManager.getLogger(KafkaConnectApiImpl.class);
+    private final Vertx vertx;
+    private final String host;
+    private final int port;
+
+    public KafkaConnectApiImpl(Vertx vertx, String host, int port) {
+        this.vertx = vertx;
+        this.host = host;
+        this.port = port;
+    }
+
+    @Override
+    public Future<JsonObject> createOrUpdatePutRequest(String connectorName,
+                                                       JsonObject configJson) {
+        Future<JsonObject> result = Future.future();
+        HttpClientOptions options = new HttpClientOptions().setLogActivity(true);
```

Comment:
I would use Vert.x Web Client. It can deal with JSON object directly without passing through a buffer. Even handling response is much simpler. I used that for testing the bridge. More details here: https://vertx.io/docs/vertx-web-client/java/

Reply:
Using HTTP or Web Client, we could have just one member instance of it, instead of creating a new one for each request.

Reply:
@pamelipluas used the Web Client initially, but TBH it wasn't much simpler. I felt that it wasn't really worth adding another dependency to the CO just for something that was only _very slightly_ simpler.

Reply:
"it wasn't much simpler", really? 
Btw fine about the dependency reason.

Reply:
_Really_, the difference was a few lines. Maybe I'm just more of a low level kind of a person, but I didn't see any extra value.

Reply:
Don't talk me about "low level" ... in the old days I didn't have any HTTP clients to use on my embedded system and I used to develop a custom one writing on a TCP socket :P

Reply:
I wasn't playing low-level top trumps, merely saying that in this case at least the supposed higher level api of vertx-web wasn't of value to me. :slightly_smiling_face:

Reply:
Isn't one of these planned to be removed in the future Vert.x versions?

Reply:
not sure about that, where did you hear about it? Let me ask Vert.x people.

Reply:
The deprecation warnings I saw only said that one of the methods was changing from `Handler<Foo>` to `Handler<AsyncResult<Foo>>`, which would require only a minor code change.

Reply:
Asked ... they will be not removed at all but for Vert.x 4, HTTP client will have some breaking changes in the API.